src/online2bin/online2-wav-nnet3-latgen-faster.cc

```diff
@@ -0,0 +1,288 @@
+        "(nnet3 setup), with optional iVector-based speaker adaptation and\n"
+        "optional endpointing.  Note: some configuration values and inputs are\n"
+        "set via config files whose filenames are passed as options\n"
+        "\n"
+        "Usage: online2-wav-nnet3-latgen-faster [options] <nnet3-in> <fst-in> "
+        "<spk2utt-rspecifier> <wav-rspecifier> <lattice-wspecifier>\n"
+        "The spk2utt-rspecifier can just be <utterance-id> <utterance-id> if\n"
+        "you want to decode utterance by utterance.\n";
+    
+    ParseOptions po(usage);
+    
+    std::string word_syms_rxfilename;
+    
+    OnlineEndpointConfig endpoint_config;
+
+    // feature_config includes configuration for the iVector adaptation,
+    // as well as the basic features.
+    OnlineNnet2FeaturePipelineConfig feature_config;
+    OnlineNnet3DecodingConfig nnet3_decoding_config;
+
+    BaseFloat chunk_length_secs = 0.18;
+    bool do_endpointing = false;
+    bool online = true;
+    
+    po.Register("chunk-length", &chunk_length_secs,
+                "Length of chunk size in seconds, that we process.  Set to <= 0 "
+                "to use all input in one chunk.");
+    po.Register("word-symbol-table", &word_syms_rxfilename,
+                "Symbol table for words [for debug output]");
+    po.Register("do-endpointing", &do_endpointing,
+                "If true, apply endpoint detection");
+    po.Register("online", &online,
+                "You can set this to false to disable online iVector estimation "
+                "and have all the data for each utterance used, even at "
+                "utterance start.  This is useful where you just want the best "
+                "results and don't care about online operation.  Setting this to "
+                "false has the same effect as setting "
+                "--use-most-recent-ivector=true and --greedy-ivector-extractor=true "
+                "in the file given to --ivector-extraction-config, and "
+                "--chunk-length=-1.");
+    po.Register("num-threads-startup", &g_num_threads,
+                "Number of threads used when initializing iVector extractor.");
+    
+    feature_config.Register(&po);
+    nnet3_decoding_config.Register(&po);
+    endpoint_config.Register(&po);
+    
+    po.Read(argc, argv);
+    
+    if (po.NumArgs() != 5) {
+      po.PrintUsage();
+      return 1;
+    }
+    
+    std::string nnet3_rxfilename = po.GetArg(1),
+        fst_rxfilename = po.GetArg(2),
+        spk2utt_rspecifier = po.GetArg(3),
+        wav_rspecifier = po.GetArg(4),
+        clat_wspecifier = po.GetArg(5);
+    
+    OnlineNnet2FeaturePipelineInfo feature_info(feature_config);
+
+    if (!online) {
+      feature_info.ivector_extractor_info.use_most_recent_ivector = true;
+      feature_info.ivector_extractor_info.greedy_ivector_extractor = true;
+      chunk_length_secs = -1.0;
+    }
+    
+    TransitionModel trans_model;
+    nnet3::AmNnetSimple nnet;
```

Comment:
For consistency with rest of nnet3 code, this variable and its derivatives could be renamed am_nnet.